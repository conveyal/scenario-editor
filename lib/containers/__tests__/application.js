/* global describe, expect, it */

describe('Container > Application', () => {
  const nock = require('nock')
  const React = require('react')
  const {Provider} = require('react-redux')
  const renderer = require('react-test-renderer')
  const {makeMockStore, mockStores} = require('../../utils/mock-data')

  it('renders correctly', () => {
    const Application = require('../application')
    const mockStore = makeMockStore(mockStores.init)

    // ensure that /api/project is called during first load
    nock('http://mockHost.com/')
      .get('/api/project')
      .reply(200, {})

    const props = {
      // react-router params
      location: {
        pathname: '/'
      },
      params: {}
    }

    // mount component
    const component = renderer.create(
      <Provider store={mockStore}>
        <Application
          {...props}
          />
      </Provider>
    )
    // when trying to make a snapshot from mountToJson, I got an out of memory error

    const tree = component.toJSON()
    // assert that application was rendered
    expect(tree).toMatchSnapshot()
  })
})
