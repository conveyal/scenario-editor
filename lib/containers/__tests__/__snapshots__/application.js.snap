// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container > Application renders correctly 1`] = `
<div
  id="editor"
>
  <div
    className="Sidebar"
  >
    <a
      className="Sidebar-logo"
      onClick={[Function]}
      style={Object {}}
      title="Projects"
    />
    <div
      className="Sidebar-bottom"
    >
      <div
        className="Sidebar-navItem"
        title="help"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          <i
            className="fa fa-question-circle fa-fw "
          />
          <span
            className="Sidebar-navItem-text"
          >
             
            help
          </span>
        </a>
      </div>
    </div>
  </div>
  <div
    className="Fullscreen  "
  >
    <div
      className={undefined}
      id="map1"
      style={undefined}
    />
  </div>
  <div
    className="ApplicationDock "
  />
</div>
`;
